datalake/src/main/java/com/sequenceiq/datalake/flow/loadbalancer/dns/UpdateLoadBalancerDNSActions.java

```diff
@@ -64,6 +75,14 @@ protected void doExecute(SdxContext context, StartUpdateLoadBalancerDNSEvent pay
                 updateLoadBalancerDNSService.performLoadBalancerDNSUpdate(sdxCluster);
                 LOGGER.info("Successfully updated the load balancer DNS for cluster {}.", sdxCluster.getClusterName());
                 eventSenderService.notifyEvent(sdxCluster, context, ResourceEvent.UPDATE_LOAD_BALANCER_DNS_FINISHED);
+                // if the flow is the resize recover we should put a new event which is the recovery finished
+                Optional<FlowLogWithoutPayload> lastFlowLog = flowLogService.getLastFlowLog(context.getFlowParameters().getFlowId());
+                if (lastFlowLog.isPresent()) {
+                    Optional<FlowChainLog> flowChainLog = flowChainLogService.findFirstByFlowChainIdOrderByCreatedDesc(lastFlowLog.get().getFlowChainId());
+                    if (flowChainLog.isPresent() && flowChainLog.get().getFlowChainType().equals(DatalakeResizeFlowEventChainFactory.class.getSimpleName())) {
+                        eventSenderService.sendEventAndNotification(sdxCluster, context.getUserId(), ResourceEvent.DATALAKE_RECOVERY_FINISHED);
+                    }
+                }
```

Comment:
you could just drop the comment and move this to a well named method like `sendNotificationInCaseOfResizeRecovery`